waitress/adjustments.py

```diff
@@ -324,6 +367,56 @@ def __init__(self, **kw):
             except:
                 raise ValueError('Invalid host/port specified.')
 
+        if (
+            self.trusted_proxy is None and
+            (
+                self.trusted_proxy_headers or
+                (self.clear_untrusted_proxy_headers is not _bool_marker)
+            )
+        ):
+            raise ValueError(
+                "The values trusted_proxy_headers and clear_untrusted_proxy_headers "
+                "have no meaning without setting trusted_proxy. Cowardly refusing to "
+                "continue."
```

Comment:
Suggest something like this:

trusted_proxy must be set when trusted_proxy_headers and clear_untrusted_proxy_headers are set.